docs/imageMgmtAPI.rst

```diff
@@ -0,0 +1,373 @@
+  to let Azkaban know the state of the image.
+* **DEPRECATED** - Any version which is unused or no longer required can be marked as ``DEPRECATED`` using
+  the `update API <#update-image-version>`_.
+
+**Response:**
+
+.. code-block::
+
+     Status: 201 Created
+     Header -> Location: /imageVersions/{id}
+
+.. _get-image-version:
+
+Get Image Version Metadata
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+This API gets the image metadata for an image type. This is typically a search API. In addition to mandatory
+parameter ``imageType``, the API takes two optional parameters: ``imageVersion`` and ``versionState``. Based
+on these parameters we should be able to perform search on top of all the available/registered image versions
+for an image type. To get the metadata for a specific image version, the Optional parameter: **imageVersion**
+should be populated in the request structure. Similarly, to filter the versions based on state, the caller
+should populate the optional parameter: **versionState**.
+
+- **Method:** GET
+- **Request URL:** /imageVersions?imageType=<image_type>
+- **Request Body:**
+
++-----------------+-------------+-------------------------------------------------+
+|   Field Name    |     Type    |            Description                          |
++=================+=============+=================================================+
+| ``imageType``   | ``String``  |                                                 |
++-----------------+-------------+-------------------------------------------------+
+| ``imageVersion``| ``String``  | OPTIONAL Parameter.                             |
++-----------------+-------------+-------------------------------------------------+
+| ``versionState``| ``String``  | OPTIONAL Parameter.                             |
++-----------------+-------------+-------------------------------------------------+
+
+**Example:**
+
+.. code-block::
+
+     GET /imageVersions?imageType=spark_job&imageVersion=1.1.0
+
+**Response:**
+
+.. code-block::
+
+     Status: 200 OK
+     Returns the matching records
+     Format: json
+
+.. _update-image-version:
+
+Update Image Version Metadata
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Update image version metadata such as state. Possible values for ``state`` are:
+``new``, ``active``, ``unstable`` or “deprecated” etc. The state will be updated during the process of ramp up.
+
+- **Method:** PATCH
+- **Request URL:** /imageVersions/{imageType}
+- **Request Body:**
+
++------------------+-------------+-----------------------------------------------------+
+|    Field Name    |     Type    |            Description                              |
++==================+=============+=====================================================+
+| ``versionState`` | ``String``  | ``NEW``, ``ACTIVE``, ``UNSTABLE`` or ``DEPRECATED`` |
++------------------+-------------+-----------------------------------------------------+
```

Comment:
Example payload for create version:

{
    "imageType": "spark",
    "imagePath": "updated full image path",
    "versionState": "ACTIVE"
}